add an orchestrate to upgrade the workload plane node to a control plane node

charlesprost opened the pull request:
Execute this orchestrate after the one from #735 and you'll have your new node integrated into the control plane.

For this, execute on the salt-master container:
```
salt-run state.orch metalk8s.orchestrate.control_plane saltenv=metalk8s-2.0 pillar='{"bootstrap_id": "bootstrap", "node_name": "node1"}'
```

Then execute on the bootstrap node:

```
kubectl -n kube-system get pods -o wide|grep node1
kubectl -n kube-system get nodes node1
```

Make sure you have 5 pods running on node1, node status is `Ready` and node role is `master`.

Issues to solve/to check:

* Make sure the node1 apiserver pod is targeting the bootstrap etcd pod
* Make sure the node1 kubelet is targeting the node1 apiserver pod

NicolasT commented on salt/metalk8s/orchestrate/control_plane.sls:
```diff
@@ -0,0 +1,48 @@
+    - sls:
+          - metalk8s.kubeadm.init.certs.apiserver
+          - metalk8s.kubeadm.init.certs.apiserver-kubelet-client
+          - metalk8s.kubeadm.init.certs.sa
```
Not sure this'll work. I believe the ServiceAccount key-pair needs to be *shared* across all `k-c-m` nodes, it's not really a CA-based system. To be validated, though.

charlesprost replied:
Fixed. Now we deploy the SA CA cert everywhere.

NicolasT commented on salt/metalk8s/orchestrate/control_plane.sls:
```diff
@@ -0,0 +1,48 @@
+    - tgt: {{ node }}
+    - saltenv: {{ saltenv }}
+    - sls:
+      - metalk8s.bootstrap.mark_control_plane
```
This should run on the master, not on the node.

NicolasT replied:
It won't even work on the node: node shouldn't have a `kubeconf`, definitely not an `admin.conf`.

charlesprost replied:
Right now we run it on the bootstrap node since it has all what is necessary but ultimately we want it run on the master indeed. See debt issue #910

NicolasT commented on salt/metalk8s/orchestrate/control_plane.sls:
```diff
@@ -0,0 +1,48 @@
+    - saltenv: {{ saltenv }}
+    - sls:
+      - metalk8s.bootstrap.mark_control_plane
+      - metalk8s.bootstrap.addons
```
This is not required to re-run when adding a node. The `Deployment`s and `DaemonSet`s are created once.

charlesprost replied:
Removed.

NicolasT commented on salt/metalk8s/orchestrate/control_plane.sls:
```diff
@@ -0,0 +1,48 @@
+    - sls:
+      - metalk8s.bootstrap.mark_control_plane
+      - metalk8s.bootstrap.addons
+      - metalk8s.bootstrap.calico
```
Same as https://github.com/scality/metalk8s/pull/886/files#r270109384, except maybe `calico-cni-plugin` needs to be installed, but that should already be handled by the 'first stage' of node addition.

charlesprost replied:
Removed.

charlesprost committed:
deployment: make mark_control_plane state generic

Issue: #736

We need to be able to configure a specific node even while not running the state on it.

charlesprost committed:
deployment: add an orchestrate to add a node into control plane

Issue: #736

This orchestrate can be executed from the salt-master container:

    salt-run state.orch metalk8s.orchestrate.control_plane saltenv=metalk8s-2.0 pillar='{"bootstrap_id": "bootstrap", "node_name": "node1"}'

charlesprost committed:
deployment: deploy etcd CA certificate from the mine when necessary

Issue: #736

In order to deploy an APIserver on a new node, we need to have the etcd/ca.crt deployed.

charlesprost committed:
deployment: add SA pub key in the mine and deploy it when necessary

Issue: #736

In order to deploy an APIserver on a new node, we need to have the sa.pub deployed.

charlesprost committed:
deployment: deploy front-proxy CA certificate from the mine when necessary

Issue: #736

In order to deploy an APIserver on a new node, we need to have the front-proxy-ca.crt
deployed.

charlesprost committed:
deployment: make apiserver state generic

Issue: #736

We need to be able to point at the etcd running on the bootstrap node.

ramanduh commented on salt/metalk8s/orchestrate/control_plane.sls:
```diff
@@ -0,0 +1,65 @@
+      - salt: Update mine
+    - pillar:
+        repo:
+          online_mode: false
```
we don't need to override the 2 values: online_mode and local_mode

charlesprost replied:
Thx done.

ramanduh commented on salt/metalk8s/kubeadm/init/mark-control-plane/configured.sls:
```diff
@@ -1,4 +1,4 @@
-{% set hostname = salt['network.get_hostname']() %}
+{% set hostname = salt['network.get_hostname']() if 'mark_control_plane_hostname' not in pillar.keys() else pillar['mark_control_plane_hostname'] %}
```
could be simplified by:
```
pillar.get('mark_control_plane_hostname', salt['network.get_hostname']())
```

charlesprost replied:
Thx, doneE.

TeddyAndrieux commented on salt/metalk8s/orchestrate/control_plane.sls:
```diff
@@ -0,0 +1,65 @@
+    - tgt: {{ node }}
+    - saltenv: {{ saltenv }}
+    - sls:
+      - metalk8s.mine.ips
```
You will need to remove it before merge it to `development/2.0` because I removed this state in #909

bert-e commented:
# Hello nootal,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/orchestrate-runner-to-bootstrap-control-plane-node` into
`development/2.0`
* [w/2.1/improvement/orchestrate-runner-to-bootstrap-control-plane-node](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/orchestrate-runner-to-bootstrap-control-plane-node) will be merged into `development/2.1` 
* [w/2.2/improvement/orchestrate-runner-to-bootstrap-control-plane-node](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/orchestrate-runner-to-bootstrap-control-plane-node) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf





The following reviewers are expecting changes from the author, or must review again:

* @teddyandrieux

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf

sayf-eddine-scality reviewed (dismissed):
lgtm

charlesprost commented:
@bert-e approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/944c2c34988a79d3fcaacc89d4ca043717bb55b7) did not succeed in branch w/2.2/improvement/orchestrate-runner-to-bootstrap-control-plane-node.




*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/544a0f45d741fd61b00b95c16ccf64a2df195e0f) did not succeed in branch improvement/orchestrate-runner-to-bootstrap-control-plane-node.




*The following options are set:* **approve**

bert-e commented:
# History mismatch



Merge commit #ff0d6c49a6a991d58987928fe2c04d2be41455a4 on the integration branch
`w/2.1/improvement/orchestrate-runner-to-bootstrap-control-plane-node` is merging a branch which is neither the current
branch `improvement/orchestrate-runner-to-bootstrap-control-plane-node` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/orchestrate-runner-to-bootstrap-control-plane-node` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

charlesprost commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/orchestrate-runner-to-bootstrap-control-plane-node` into
`development/2.0`
* [w/2.1/improvement/orchestrate-runner-to-bootstrap-control-plane-node](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/orchestrate-runner-to-bootstrap-control-plane-node) will be merged into `development/2.1` 
* [w/2.2/improvement/orchestrate-runner-to-bootstrap-control-plane-node](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/orchestrate-runner-to-bootstrap-control-plane-node) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf










*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/d3154eeeef0f9c44c69b67320c9957dacf09633d) did not succeed in branch improvement/orchestrate-runner-to-bootstrap-control-plane-node.




*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf










*The following options are set:* **approve**

charlesprost committed:
CI: restrict control and workload plane networks on the single node env to avoid confusion while picking IPs

Fixes #920

Our ranges were too wide and could include other IPs that were not part of those networks.
By narrowing them down, we are sure we pick the right IP.

bert-e commented:
# History mismatch



Merge commit #851fc02182463ea20949f131a725880970197130 on the integration branch
`w/2.1/improvement/orchestrate-runner-to-bootstrap-control-plane-node` is merging a branch which is neither the current
branch `improvement/orchestrate-runner-to-bootstrap-control-plane-node` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/orchestrate-runner-to-bootstrap-control-plane-node` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

bert-e commented:
# History mismatch



Merge commit #19c8b4420e2e581f4784040d316c79590e731415 on the integration branch
`w/2.1/improvement/orchestrate-runner-to-bootstrap-control-plane-node` is merging a branch which is neither the current
branch `improvement/orchestrate-runner-to-bootstrap-control-plane-node` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/orchestrate-runner-to-bootstrap-control-plane-node` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

charlesprost commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/orchestrate-runner-to-bootstrap-control-plane-node` into
`development/2.0`
* [w/2.1/improvement/orchestrate-runner-to-bootstrap-control-plane-node](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/orchestrate-runner-to-bootstrap-control-plane-node) will be merged into `development/2.1` 
* [w/2.2/improvement/orchestrate-runner-to-bootstrap-control-plane-node](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/orchestrate-runner-to-bootstrap-control-plane-node) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf










*The following options are set:* **approve**

bert-e commented:
# History mismatch



Merge commit #fa8c5868c7d25213e7cba496a4bbf2cf2e74ba4e on the integration branch
`w/2.1/improvement/orchestrate-runner-to-bootstrap-control-plane-node` is merging a branch which is neither the current
branch `improvement/orchestrate-runner-to-bootstrap-control-plane-node` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `improvement/orchestrate-runner-to-bootstrap-control-plane-node` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

charlesprost commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/orchestrate-runner-to-bootstrap-control-plane-node` into
`development/2.0`
* [w/2.1/improvement/orchestrate-runner-to-bootstrap-control-plane-node](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/improvement/orchestrate-runner-to-bootstrap-control-plane-node) will be merged into `development/2.1` 
* [w/2.2/improvement/orchestrate-runner-to-bootstrap-control-plane-node](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/improvement/orchestrate-runner-to-bootstrap-control-plane-node) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf










*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



Please check the status of the associated issue None.

Goodbye nootal.
